sql/core/src/test/scala/org/apache/spark/sql/CharVarcharTestSuite.scala

```diff
@@ -56,7 +56,7 @@ trait CharVarcharTestSuite extends QueryTest with SQLTestUtils {
       checkAnswer(spark.table("t"), Row("1", "a" + " " * 4))
       checkColType(spark.table("t").schema(1), CharType(5))
 
-      sql("ALTER TABLE t DROP PARTITION(c='a')")
+      sql("ALTER TABLE t DROP PARTITION(c='a    ')")
```

Comment:
is it the same with hive? or we should also add padding for char type column in partition spec?

Reply:
Not same now, we should add padding to this, as for char(4), 'a', 'a ', 'a  ', 'a   ' are all same